src/main/java/de/feu/massim22/group3/agents/Desires/Desires.java

```diff
@@ -0,0 +1,50 @@
+package de.feu.massim22.group3.agents.Desires;
+
+public enum Desires {
+    // Team Desires
+    PROCESS_TASK,
+    HELP_TASK_AGENT,
+    EXPLORE_MAP,
+    DETERMINE_MAP_SIZE,
+    ATTACK_ENEMY,
+    GOAL_ZONE_GUARD,
+    // Agent Desires
+    DIG_FREE,
+    DODGE_CLEAR,
+    DODGE_OTHER_AGENT,
+    REACT_TO_NORM,
+    SPONTANEOUS_HINDER_ENEMY,
+    GO_TO_UNKNOWN_AREA,;
+
+    public Desire getDesireObj() {
+        switch (this) {
+        case PROCESS_TASK:
+            return new ProcessTaskDesire(PROCESS_TASK);
```

Comment:
Das Enum in den Konstruktor zu übergeben, schaut auf den ersten Blick wegen der duplizierten Infos etwas komisch aus. Alternativ könnte man eine setType Methode in Desire ergänzen, die vom Konstruktor aufgerufen wird und die von der jeweiligen Subklasse implementiert werden muss. Ist aber Geschmackssache.